src/inmanta/ast/statements/assign.py

```diff
@@ -159,6 +166,9 @@ def execute(self, requires: typing.Dict[object, object], resolver: Resolver, que
 
         return qlist
 
+    def validate_as_default_attribute(self, expected_type: "Type", multi: bool = False, nullable: bool = False) -> None:
```

Comment:
the method name could be more clear
suggestions

- assert_type
- ensure_type
- check_type_is
- check_type_for_constant